ahnlich/ai/src/server/task.rs

```diff
@@ -153,27 +155,35 @@ impl AhnlichProtocol for AIProxyTask {
                 }
 
                 AIQuery::DelKey { store, key } => {
-                    let default_metadatakey = &*AHNLICH_AI_RESERVED_META_KEY;
-                    let metadata_value: MetadataValue = key.into();
-                    let delete_condition = PredicateCondition::Value(Predicate::In {
-                        key: default_metadatakey.clone(),
-                        value: HashSet::from_iter([metadata_value]),
-                    });
-
-                    match self
-                        .db_client
-                        .del_pred(store, delete_condition, parent_id.clone())
-                        .await
-                    {
-                        Ok(res) => {
-                            if let ServerResponse::Del(num) = res {
-                                Ok(AIServerResponse::Del(num))
-                            } else {
-                                Err(AIProxyError::UnexpectedDBResponse(format!("{:?}", res))
-                                    .to_string())
+                    match self.store_handler.store_original(store.clone()) {
+                        Err(err) => Err(err.to_string()),
+                        Ok(false) => Err(AIProxyError::DelKeyError.to_string()),
```

Comment:
If we can't delete key for cases when `store_original` is `false`, how do we get to delete such inputs? Will they be stuck in the db forever? Or can we currently then delete via their metadata?

Reply:
We as the server can't because we don't track any other metadata entry... but the user can

Reply:
For instance if the user decides to introduce a metadatavalue entry of (key, uuid) that's always guaranteed to be the same per entry then yes they can delete that ... perhaps that points to having a more atomic `SetPred` command where you can set that key where the predicate suffices

Reply:
What command will they use to delete in that case?

Reply:
They can use `DelPred`